framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -4631,6 +4613,442 @@ private void makeGroundTargetType(
     addDefaultAnnotations(functionalType);
   }
 
+  /**
+   * Return true if {@code type} should be captured.
+   *
+   * <ul>
+   *   <li>{@code type} and {@code typeMirror} must both be declared types.
+   *   <li>{@code type} must not be a raw type, must not have an uninferred type argument, and must
+   *       have a wildcard as a type argument.
+   *   <li>{@code typeMirror} must have a captured type as a type argument.
+   * </ul>
+   *
+   * @param type annotated type to capture
+   * @param typeMirror underlying Java type
+   * @return true if {@code type} should be captured
+   */
+  private boolean shouldCapture(AnnotatedTypeMirror type, TypeMirror typeMirror) {
+    if (type.getKind() != TypeKind.DECLARED) {
+      return false;
+    }
+    if (typeMirror.getKind() != TypeKind.DECLARED) {
+      throw new BugInCF("Expected declared type mirror: %s, type: %s", typeMirror, type);
+    }
+
+    DeclaredType capturedTypeMirror = (DeclaredType) typeMirror;
+    AnnotatedDeclaredType uncapturedType = (AnnotatedDeclaredType) type;
+    if (capturedTypeMirror.getTypeArguments().isEmpty()) {
+      return false;
+    }
+
+    if (uncapturedType.wasRaw() || uncapturedType.containsUninferredTypeArguments()) {
+      return false;
+    }
+
+    if (capturedTypeMirror.getTypeArguments().size() != uncapturedType.getTypeArguments().size()) {
+      throw new BugInCF(
+          "Not the same number of type arguments: capturedTypeMirror: %s uncapturedType: %s",
+          capturedTypeMirror, uncapturedType);
+    }
+
+    for (int i = 0; i < capturedTypeMirror.getTypeArguments().size(); i++) {
+      TypeMirror capturedTypeArgTM = capturedTypeMirror.getTypeArguments().get(i);
+      AnnotatedTypeMirror uncapturedTypeArg = uncapturedType.getTypeArguments().get(i);
+      if (uncapturedTypeArg.getKind() != TypeKind.WILDCARD) {
+        continue;
+      }
+      if (TypesUtils.isCaptured(capturedTypeArgTM)
+          || capturedTypeArgTM.getKind() != TypeKind.WILDCARD) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Apply capture conversion.
+   *
+   * <p>Capture conversion is the process of converting wildcards in a generic type to fresh type
```

Comment:
"generic" always refers to a type declaration, which has type variables, not type arguments.
I think that "generic" here should be "parameterized".  The same comment applies to the other overload of this method.